src/AutoRest.CSharp/Mgmt/Output/Samples/MgmtSampleProvider.cs

```diff
@@ -463,9 +472,29 @@ private MethodBodyStatement BuildGetCollectionStatement(ResourceCollection colle
             }
 
             // call the method to get the collection instance
-            statements.Add(
-                Declare(collection.Type, "collection", parentVar.Invoke(getResourceCollectionMethodName, arguments), out instance)
-                );
+            if (parentVar == null)
+            {
+                // Can't use CSharpType.Equals(typeof(...)) because the CSharpType.Equals(Type) would assume itself is a FrameworkType, but here it's generated when IsArmCore=true
+                if (Configuration.MgmtConfiguration.IsArmCore && parent.Type.Name == nameof(ArmResource))
+                {
+                    // use GenericResource to get the collection
+                    statements.Add(
+                        Declare(collection.Type, "collection", client.Invoke("GetGenericResource", arguments[0]).Invoke(getResourceCollectionMethodName, arguments[1..]), out instance)
+                        );
+                }
+                else
```

Comment:
This is a bit confusing, we set parentVar to null if parent is another ArmResource.
But here we need to handle the parent is not ArmResource when parentVar is null?

Reply:
I will try to reorganize this if this confuses people